build_tools/cmake/iree_bytecode_module.cmake

```diff
@@ -42,7 +42,7 @@ function(iree_bytecode_module)
     _RULE
     "PUBLIC;TESTONLY"
     "NAME;SRC;MODULE_FILE_NAME;COMPILE_TOOL;C_IDENTIFIER;FRIENDLY_NAME;STATIC_LIB_PATH"
-    "FLAGS;DEPENDS;DEPS"
+    "FLAGS;DEPENDS;DEPS;DATA"
```

Comment:
Please add a mention of this new `DATA` parameter in the function comment above. Should mention that it's currently CMake-only (not available in the Bazel function and not handled in `bazel_to_cmake_converter.py`), ideally file an Issue about that and reference it in a `TODO(#12345)`.

Reply:
As discussed, it seems this is a compile-time dependency of the module, not a runtime dependency of the test. So this should be "deps". Except that in Bazel we've already overloaded "deps" to mean dependencies of the generated `cc_library`. I suspect that's why this CMake rule also has `DEPENDS`. If you look under the hood, that's all that `iree_add_data_dependencies` is really doing for you except it's also trying (poorly) to determine if something is a target. So if you threaded your bitcode library into this `DEPENDS` argument you wouldn't need `DATA` at all and that would be more correct because this isn't really a runtime data dependency. Part of the issue comes from the different model between Bazel and CMake wrt configurations and part comes because we're hacking together macros rather than building a real set of Bazel rules (like `cc_library`). And we've tagged the `cc_embed_data` onto `iree_bytecode_module` in a way that their arguments get all mixed up.

Reply:
I am not sure that this works. I changed to use this, and I get the following error 
```
[build] [0/2] Re-checking globbed directories...
[build] ninja: error: 'samples/custom_dispatch/cuda/kernels/cuda_ukernel', needed by 'samples/custom_dispatch/cuda/kernels/check_cuda_ukernel_ukernel_example.mlir_module.vmfb', missing and no known rule to make it
[proc] The command: /usr/bin/cmake --build /usr/local/google/home/ravishankarm/iree/build/Debug --parallel 130 --target all exited with code: 1
[driver] Build completed: 00:00:01.359
[build] Build finished with exit code 1
```